app/src/sharedTest/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentTest.kt

```diff
@@ -1205,6 +1211,92 @@ class TopicLessonsFragmentTest {
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicLessonsFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicLessonsFragment
+
+        val receivedInternalProfileId = topicLessonsFragment
+          .arguments?.extractCurrentUserProfileId()?.internalId ?: -1
+        val args = topicLessonsFragment.arguments?.getProto(
+          TopicLessonsFragment.TOPIC_LESSONS_FRAGMENT_ARGUMENTS_KEY,
+          TopicLessonsFragmentArguments.getDefaultInstance()
+        )
+        val receivedClassroomId = checkNotNull(args?.classroomId) {
+          "Expected classroom ID to be included in arguments for TopicLessonsFragment."
+        }
+        val receivedTopicId = checkNotNull(args?.topicId) {
+          "Expected topic ID to be included in arguments for TopicLessonsFragment."
+        }
+        val receivedStoryId = args?.storyId ?: ""
+
+        assertThat(receivedInternalProfileId).isEqualTo(internalProfileId)
+        assertThat(receivedClassroomId).isEqualTo(TEST_CLASSROOM_ID_1)
+        assertThat(receivedTopicId).isEqualTo(RATIOS_TOPIC_ID)
+        assertThat(receivedStoryId).isEqualTo(RATIOS_STORY_ID_0)
+      }
+    }
+  }
+
+  @Test
+  fun testTopicLessonsFragment_saveInstanceState_verifyCorrectStateRestored() {
+    launch<TopicActivity>(
+      createTopicPlayStoryActivityIntent(
+        internalProfileId,
+        TEST_CLASSROOM_ID_1,
+        RATIOS_TOPIC_ID,
+        RATIOS_STORY_ID_0
+      )
+    ).use { scenario ->
+      clickLessonTab()
+      testCoroutineDispatchers.runCurrent()
+      scenario.onActivity { activity ->
+        val topicFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.topic_fragment_placeholder) as TopicFragment
+        val viewPager = topicFragment.requireView()
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicLessonsFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicLessonsFragment
+
+        topicLessonsFragment.onExpandListIconClicked(1)
+      }
+
+      scenario.recreate()
+
+      scenario.onActivity { activity ->
+        val topicFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.topic_fragment_placeholder) as TopicFragment
+        val viewPager = topicFragment.requireView()
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicLessonsFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicLessonsFragment
+
+        val receivedCurrentExpandedChapterListIndex = topicLessonsFragment
+          .getCurrentExpandedChapterListIndex()
+        val receivedIsDefaultStoryExpanded = topicLessonsFragment.getIsDefaultStoryExpanded()
+
+        assertThat(receivedCurrentExpandedChapterListIndex).isEqualTo(1)
+        assertThat(receivedIsDefaultStoryExpanded).isEqualTo(true)
+      }
+    }
```

Comment:
@BenHenning ,Could we test like that without exposing internal variable of TopicLessonsFragment (currentExpandedChapterListIndex,  isDefaultStoryExpanded) ? As you mention above.

@Test
  fun testFragment_saveInstanceState_restoresCorrectly() {
    launch<TopicActivity>(
      createTopicPlayStoryActivityIntent(
        internalProfileId,
        TEST_CLASSROOM_ID_1,
        RATIOS_TOPIC_ID,
        RATIOS_STORY_ID_0
      )
    ).use { scenario ->
      clickLessonTab()
      testCoroutineDispatchers.runCurrent()

      scrollToPosition(position = 2)
      clickStoryItem(position = 2, targetViewId = R.id.chapter_list_drop_down_icon)

      scenario.recreate()

      scrollToPosition(position = 2)
      onView(
        atPositionOnView(
          recyclerViewId = R.id.story_summary_recycler_view,
          position = 2,
          targetViewId = R.id.chapter_recycler_view
        )
      ).check(matches(isDisplayed()))
    }
  }

Reply:
I think this approach of verifying looks really reasonable!